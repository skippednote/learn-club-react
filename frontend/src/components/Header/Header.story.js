import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { storiesOf } from '@storybook/react';
import Header from './index';

storiesOf('Header', module).add('default', () => (
  <MemoryRouter>
    <Header />
  </MemoryRouter>
));
