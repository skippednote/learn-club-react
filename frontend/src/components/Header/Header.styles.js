import styled from 'react-emotion';

export const Header = styled('header')`
  text-align: center;
`;
Header.displayName = 'Header';

export const Menu = styled('ul')`
  width: 300px;
  margin: 0 auto 2rem;
  list-style: none;
  padding-left: 0;
  display: flex;
  justify-content: space-evenly;
`;
Menu.displayName = 'Menu';

export const MenuItem = styled('li')``;
MenuItem.displayName = 'MenuItem';
