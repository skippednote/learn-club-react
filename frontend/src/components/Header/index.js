// @flow
import React from 'react';
import Logo from '../Logo';

const Header = () => (
  <header>
    <Logo>âš›</Logo>
    <h1>Learn Club - React</h1>
  </header>
);

export default Header;
