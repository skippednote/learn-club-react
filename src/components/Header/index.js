// @flow
import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../Logo';
import { Menu, MenuItem, Header as Wrapper } from './Header.styles';

const Header = () => (
  <Wrapper>
    <Link to="/">
      <Logo>âš›</Logo>
    </Link>
    <h1>Learn Club - React</h1>
    <Menu>
      <MenuItem>
        <Link to="/">Home</Link>
      </MenuItem>
      <MenuItem>
        <Link to="/people">People</Link>
      </MenuItem>
      <MenuItem>
        <Link to="/articles">Articles</Link>
      </MenuItem>
    </Menu>
  </Wrapper>
);

export default Header;
