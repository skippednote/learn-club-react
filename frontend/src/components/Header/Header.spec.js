import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from './index';

describe('Counter', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Header />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the Logo', () => {
    const wrapper = shallow(<Header />);
    const logo = wrapper.find('Logo');
    expect(logo).toHaveLength(1);
  });

  it('renders the title', () => {
    const wrapper = shallow(<Header />);
    const title = wrapper.find('h1');
    expect(title).toHaveLength(1);
    expect(title.text()).toContain('Learn Club - React');
  });
});
