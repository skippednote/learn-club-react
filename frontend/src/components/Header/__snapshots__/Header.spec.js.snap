// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches snapshot 1`] = `
<header>
  <Logo>
    âš›
  </Logo>
  <h1>
    Learn Club - React
  </h1>
</header>
`;
